apps/examples/wifi_manager/wm_test/wm_test_connectbyrssi_test.c

```diff
@@ -0,0 +1,311 @@
+	apconfig.ssid_length = strlen(ap_info->ssid);
+	apconfig.ssid[WIFIMGR_SSID_LEN] = '\0';
+	apconfig.ap_auth_type = ap_info->auth_type;
+	if (ap_info->auth_type != WIFI_MANAGER_AUTH_OPEN) {
+		strncpy(apconfig.passphrase, ap_info->password, WIFIMGR_PASSPHRASE_LEN);
+		apconfig.passphrase[WIFIMGR_PASSPHRASE_LEN] = '\0';
+		apconfig.passphrase_length = strlen(ap_info->password);
+		apconfig.ap_crypto_type = ap_info->crypto_type;
+	} else{
+		apconfig.passphrase[0] = '\0';
+		apconfig.passphrase_length = 0;
+		apconfig.ap_crypto_type = ap_info->crypto_type;
+	}
+
+	_wt_print_wifi_ap_profile(&apconfig, "Connecting AP Info");
+
+	wifi_manager_result_e res = wifi_manager_connect_ap(&apconfig);
+	if (res != WIFI_MANAGER_SUCCESS) {
+		WT_LOGE(TAG, "AP connect failed");
+		return;
+	}
+	/* Wait for DHCP connection */
+	WT_LOG(TAG, "wait join done");
+	WIO_TEST_WAIT;
+	WT_LEAVE;
+}
+
+static void _wt_disconnect(void *arg)
+{
+	WT_ENTER;
+	wifi_manager_result_e res = WIFI_MANAGER_SUCCESS;
+	/* Disconnect AP */
+	res = wifi_manager_disconnect_ap();
+	if (res != WIFI_MANAGER_SUCCESS) {
+		WT_LOGE(TAG, "disconnect fail (%d)", res);
+		return;
+	}
+	WIO_TEST_WAIT;
+	WT_LEAVE;
+}
+
+static void wm_run_connect_by_rssi(void *arg)
+{
+	WT_ENTER;
+	int ret = -1;
+	opt = (struct wt_options *)arg;
+	ret = wm_connect_by_rssi_init();
+	if (ret) {
+		WT_LOGE(TAG, "Failed to run interop test ");
+		return;
+	}	
+	int passed = 0, failed = 0;
+	for(int i = 0; i < opt->repeat; i++) {		
+		WT_LOG(TAG, "Connection count %d", i);
+		memset(bssid, '\0', sizeof(bssid));
+		memset(connected_ap_mac, '\0', sizeof(connected_ap_mac));
+		_wt_scan(arg);		
+		_wt_connect(arg);
+				
+		if (bssid[0] == '\0' || connected_ap_mac[0] == '\0') {
+			failed++;			
+			continue;
+		}
+
+		WT_LOG(TAG, "Mac with highest rssi value in scan for given ssid %s", bssid);
+		WT_LOG(TAG, "Connected AP Mac %s", connected_ap_mac);
+		if (strncmp(bssid, connected_ap_mac, WIFIMGR_SSID_LEN) == 0) {
+			WT_LOG(TAG, "best rssi ap for given ssid is already connected");
+			passed++;
+		} else {
+			WT_LOG(TAG, "best rssi ap for given ssid is not connected");
+			failed++;
+		}
+		_wt_disconnect(arg);
+	}
+	
+	printf("============================================================================\n");
+	printf("TEST CASE:Best Rssi Connect Check\n");
+	printf("----------------------------------------------------------------------------\n");
+	printf("TotalCount - Pass - Fail\n");
+	printf("   %-11d%-8d%-5d\n", opt->repeat, passed, failed);	
+	printf("----------------------------------------------------------------------------\n");
+	printf("============================================================================\n");
+
+	wm_connect_by_rssi_deinit();
+	WT_LEAVE;
+}
+
+void wm_connectbyrssi_test(void *arg)
+{	
+	WT_ENTER;
+	int res = sem_init(&g_wio_sem, 0, 0);
+	if (res < 0)
+		WT_LOGE(TAG, "fail to initialize semaphore %d", errno);
+	wm_run_connect_by_rssi(arg);
+	sem_destroy(&g_wio_sem);
+	WT_LEAVE;
+}
```

Comment:
Please add a newline character at the end of the file.

Reply:
Done